### DISCUSSION TITLE
Reusing existing session for SDK authentication?

### DISCUSSION THREAD

**Discussion by emmett-exrx** [directus/directus#8394] [Archive]
If Directus and a custom app using the SDK share a domain, is there a way for the 3rd-party app avoid making the users log in twice (or set up an access token)? Is it possible to have the SDK reuse credentials from an existing logged in admin app session?

**Comment by joselcvarela** [comment]
Hello @emmett-exrx 
Have you tried to pass 'cookie' as mode on the constructor? https://docs.directus.io/reference/sdk/#options-mode

**Reply by emmett-exrx** [reply]
@joselcvarela Nope, I hadn't tried that. I think it, and `Directus.auth.refresh(true)` were the missing pieces.

All put together, the working solution looks like this:
```js
import { Auth, Directus, LocalStorage, AxiosTransport } from 'https://unpkg.com/@directus/sdk@9.0.0-rc.93/dist/sdk.esm.min.js';
const URL = "http://localhost:8055";

async function init() {
    let s = new LocalStorage(),
        t = new AxiosTransport(URL, s);

    let D = new Directus(URL, {
        auth: new Auth(t, s, { mode: "cookie" }),
        storage: s,
        transport: t,
    });

    try {
        await D.auth.refresh(true);
    } catch (e) {
        document.location = URL + "/admin/login";
    }

    /* App code */

}
```

Thanks!
